### ISSUE TITLE
Named block 'extra_header' not allowed inside of def 'html_header'

### ISSUE THREAD

**Issue by laodzu** [getnikola/nikola#3319]
Hi,

after updating to v8.0.3 I get this error:

```
(nikola) dzu@krikkit:~/blog (master)$ nikola build
Scanning posts....done!
[2019-11-24T11:24:07Z] ERROR: Nikola: Error loading tasks. An unhandled exception occurred.
[2019-11-24T11:24:07Z] ERROR: Nikola: mako.exceptions.CompileException: Named block 'extra_header' not allowed inside of def 'html_header' in file '/home/dzu/src/python/nikola/lib/python3.7/site-packages/nikola/data/themes/base/templates/base_header.tmpl' at line: 16 char: 5
[2019-11-24T11:24:07Z] NOTICE: Nikola: To see more details, run Nikola in debug mode (set environment variable NIKOLA_DEBUG=1) or use NIKOLA_SHOW_TRACEBACKS=1
(nikola) dzu@krikkit:~/blog (master)$ 
```

This is from the announcement mail of v8.0.3:

* Add ``extra_header`` and ``extra_footer`` blocks to templates
  (Issue #3291)

So I wonder if I have to do any additional things to complete the update or if this is a real bug.

Thanks in advance for any advice!

**Comment by Kwpolska** [comment]
Works for me with a clean demo site.

1. Try in a clean demo site, can you reproduce this error?
2. Is your version of Mako up-to-date? Try upgrading and reproducing in a clean demo site (`rm -rf cache` first)

**Comment by laodzu** [comment]
Thanks for the quick help!

Removing the cache does not help.

After upgrade Mako version is 1.1.0, but error still persists.

Will try to reproduce this on a demo site.

**Comment by laodzu** [comment]
Hm, clean demo site works.

Ok, so it is the zen theme that I use.  Steps to reproduce in the demo site:

nikola theme -i zen

and then simply change conf.py to include

THEME = "zen"

Then the error will show up.

**Comment by Kwpolska** [comment]
Turns out we can’t use a block inside a function in Mako. Which sucks. I created a PR #3320 with a fix.

**Comment by euphi** [comment]
Please re-open - I have this problem with a fresh install of nikola 8: (8.0.3)

```
(nikola8) xxxxxxxxxxxxxxxxxxxxxxx$ nikola build
Scanning posts........done!
[2020-01-11T11:48:03Z] ERROR: Nikola: Error loading tasks. An unhandled exception occurred.
[2020-01-11T11:48:03Z] ERROR: Nikola: mako.exceptions.CompileException: Named block 'extra_header' not allowed inside of def 'html_header' in file '/home/ian/Fablab/nikola8/lib/python3.7/site-packages/nikola/data/themes/base/templates/base_header.tmpl' at line: 16 char: 5
[2020-01-11T11:48:03Z] NOTICE: Nikola: To see more details, run Nikola in debug mode (set environment variable NIKOLA_DEBUG=1) or use NIKOLA_SHOW_TRACEBACKS=1

```


Before that I did (from `history`) 

```
mkdir nikola8
cd nikola8/
virtualenv .
source bin/activate
pip install Nikola[extras]
nikola init MyPrj
cd MyPrj/
vim conf.py                               # Change theme to hack and set HACK_VARIANT to "dark" - no other changes
nikola build                               # Runs fine, except warning that bootlog4 theme is used instead of hack
nikola theme -i hack
nikola build                              # Error 
```

**Comment by Kwpolska** [comment]
v8.0.4 is not released yet, and we don’t have a timeline for that release. You can easily fix your local copy by manually applying the fixes in #3320. (For some reason, though, I can’t reproduce this error with v8.0.3 and the `hack` theme.)

**Comment by pjkundert** [comment]
It occurs with the `detox` theme, too.  Nikola version: 8.0.3.
